internal/components/component.go

```diff
@@ -75,22 +83,32 @@ func (o *Option) GetServicePort() *corev1.ServicePort {
 	}
 }
 
-type PortBuilderOption func(*Option)
+func WithRBACRuleGenerator[T any](r RBACRuleGenerator[T]) PortBuilderOption[T] {
```

Comment:
TBH I don't feel confortable with everything returing "PortBuilder<Something>". For some components we don't have to build anything port related (some processors) but we need to get RBAC. I was not sure about this in the other PR but I though you were thinking about something different and didn't raise my concerns.

Reply:
I think i should just rename this to not be "PortBuilderOption" but rather "ParserOption" which is more accurate.